setup.py

```diff
@@ -1,20 +1,29 @@
 # -*- coding: utf-8 -*-
 
 from setuptools import setup, find_packages
-
+import platform
+import os
 
 with open("README.rst", encoding='utf-8') as f:
     readme = f.read()
 
 with open("LICENSE") as f:
     license = f.read()
 
-with open("requirements.txt", "r") as f:
-    required = f.read().splitlines()
+if platform.system == 'Linux' or platform.system == 'Darwin':
+    if os.system("grep avx512 /proc/cpuinfo") == 0:
```

Comment:
Please do not maintain duplicated requirements in 2 files. Just add the additional requirements in `requirements-avx512.txt`. I would also consider this an "extra" that needs to get installed separately. That can be added to `extra_requires` as 
```
extra_requires={
    'avx512': avx512_requirements
}
```
this means we'll have to check whether cpu is avx512 in the install script rather than here, which i think is fine.

Reply:
intel-tensorflow-avx512 gets substituted for tensorflow
So it's not really an extra. I wrote a script to do the substitution with one requirements.txt. Let me know if you think there are issues with this revision.

Reply:
got it. Much improved, thanks.